src/lib/io/sol-iio.c

```diff
@@ -544,12 +544,27 @@ channel_get_pure_name(struct sol_iio_channel *channel)
 {
     size_t channel_name_len;
     char *channel_pure_name;
+    char *channel_name_suffix;
     bool modified = false;
 
     channel_name_len = strlen(channel->name);
-    if (channel_name_len > 2) {
-        char *channel_name_suffix;
+    if (channel_name_len > 5) {
+        channel_name_suffix = channel->name + channel_name_len - 5;
```

Comment:
Please search the "in_intensity_both" in nodes.c, in order to support the Y(a number) part, we need to add the channel_id(multi channel) logic ?